fix(debian): fix KEYEXPIRED when installing packages

iurly opened the pull request:
fix the following error when trying to build the builder image for older debian packages:

Step 2/5 : RUN apt-get update && apt-get -y --no-install-recommends install cmake g++ git kmod libc6-dev libelf-dev make pkg-config && apt-get clean
 ---> Running in 81e16f5a1f7f
Get:1 http://security.debian.org jessie/updates InRelease [44.9 kB] Ign http://deb.debian.org jessie InRelease
Get:2 http://deb.debian.org jessie-updates InRelease [16.3 kB] Get:3 http://deb.debian.org jessie Release.gpg [1652 B] Get:4 http://deb.debian.org jessie Release [77.3 kB] Get:5 http://security.debian.org jessie/updates/main amd64 Packages [992 kB] Ign http://deb.debian.org jessie-updates InRelease Get:6 http://deb.debian.org jessie-updates/main amd64 Packages [20 B] Ign http://deb.debian.org jessie Release
Get:7 http://deb.debian.org jessie/main amd64 Packages [9098 kB] Fetched 10.2 MB in 0s (10.9 MB/s)
Reading package lists...
W: GPG error: http://deb.debian.org jessie-updates InRelease: The following signatures were invalid: KEYEXPIRED 1668891673 W: GPG error: http://deb.debian.org jessie Release: The following signatures were invalid: KEYEXPIRED 1668891673 Reading package lists...
Building dependency tree...
Reading state information...
The following extra packages will be installed:
  binutils cmake-data cpp cpp-4.9 g++-4.9 gcc gcc-4.9 git-man libarchive13
  libasan1 libatomic1 libc-dev-bin libcilkrts5 libcloog-isl4 libcurl3
  libcurl3-gnutls libelf1 liberror-perl libexpat1 libgcc-4.9-dev libgdbm3
  libglib2.0-0 libgomp1 libgssapi-krb5-2 libidn11 libisl10 libitm1
  libk5crypto3 libkeyutils1 libkrb5-3 libkrb5support0 libldap-2.4-2 liblsan0
  liblzo2-2 libmpc3 libmpfr4 libquadmath0 librtmp1 libsasl2-2
  libsasl2-modules-db libssh2-1 libssl1.0.0 libstdc++-4.9-dev libtsan0
  libubsan0 libxml2 linux-libc-dev perl perl-modules
Suggested packages:
  binutils-doc codeblocks eclipse ninja-build cpp-doc gcc-4.9-locales
  g++-multilib g++-4.9-multilib gcc-4.9-doc libstdc++6-4.9-dbg gcc-multilib
  manpages-dev autoconf automake libtool flex bison gdb gcc-doc
  gcc-4.9-multilib libgcc1-dbg libgomp1-dbg libitm1-dbg libatomic1-dbg
  libasan1-dbg liblsan0-dbg libtsan0-dbg libubsan0-dbg libcilkrts5-dbg
  libquadmath0-dbg gettext-base git-daemon-run git-daemon-sysvinit git-doc
  git-el git-email git-gui gitk gitweb git-arch git-cvs git-mediawiki git-svn
  lrzip glibc-doc krb5-doc krb5-user libstdc++-4.9-doc make-doc perl-doc
  libterm-readline-gnu-perl libterm-readline-perl-perl libb-lint-perl
  libcpanplus-dist-build-perl libcpanplus-perl libfile-checktree-perl
  liblog-message-simple-perl liblog-message-perl libobject-accessor-perl
Recommended packages:
  patch less rsync ssh-client ca-certificates libglib2.0-data shared-mime-info
  xdg-user-dirs krb5-locales libsasl2-modules xml-core netbase rename
  libarchive-extract-perl libmodule-pluggable-perl libpod-latex-perl
  libterm-ui-perl libtext-soundex-perl libcgi-pm-perl libmodule-build-perl
  libpackage-constants-perl
The following NEW packages will be installed:
  binutils cmake cmake-data cpp cpp-4.9 g++ g++-4.9 gcc gcc-4.9 git git-man
  kmod libarchive13 libasan1 libatomic1 libc-dev-bin libc6-dev libcilkrts5
  libcloog-isl4 libcurl3 libcurl3-gnutls libelf-dev libelf1 liberror-perl
  libexpat1 libgcc-4.9-dev libgdbm3 libglib2.0-0 libgomp1 libgssapi-krb5-2
  libidn11 libisl10 libitm1 libk5crypto3 libkeyutils1 libkrb5-3
  libkrb5support0 libldap-2.4-2 liblsan0 liblzo2-2 libmpc3 libmpfr4
  libquadmath0 librtmp1 libsasl2-2 libsasl2-modules-db libssh2-1 libssl1.0.0
  libstdc++-4.9-dev libtsan0 libubsan0 libxml2 linux-libc-dev make perl
  perl-modules pkg-config
0 upgraded, 57 newly installed, 0 to remove and 0 not upgraded. Need to get 60.4 MB of archives.
After this operation, 223 MB of additional disk space will be used. WARNING: The following packages cannot be authenticated!
  libgdbm3 libkeyutils1 cmake-data liblzo2-2 cmake libisl10 libcloog-isl4
  libmpfr4 libmpc3 kmod binutils cpp gcc g++ liberror-perl make pkg-config
E: There are problems and -y was used without --force-yes

nathan-b commented:
I would like to dig into this a little bit more. Is it the case that these repos are just deprecated and out-of-date so they're never going to get a new key? Or should we file a ticket against the Debian project? Because ignoring cryptographic signatures on these repositories does open us up to a supply-chain attack and is a security risk.

iurly commented:
> I would like to dig into this a little bit more. Is it the case that these repos are just deprecated and out-of-date so they're never going to get a new key? Or should we file a ticket against the Debian project? Because ignoring cryptographic signatures on these repositories does open us up to a supply-chain attack and is a security risk.

Not sure that's a real answer to your questions and concerns, but I came across the following statement on https://hub.docker.com/_/debian:

> If you find yourself needing a Debian release which is EOL (and thus only available from [archive.debian.org](http://archive.debian.org/)), you should check out [the debian/eol image](https://hub.docker.com/r/debian/eol/), which includes tags for Debian releases as far back as Potato (Debian 2.2), the first release to fully utilize APT.

So I tried replacing `FROM ubuntu:jessie` with `FROM ubuntu/eol:jessie` and the issue seems to have disappeared.
I'm not quite sure what the bottom line is, but I think that solution should be good enough. What do you think?

iurly committed:
fix(debian): prevent KEYEXPIRED messages by using debian/eol

So far we've been using debian:jessie (which is EOL) from the
official debian registry on docker.io.
However, as per their recommendation, we should use debian/eol
instead.

This fixes the following error when trying to build the builder image
for older debian kernels (compiled with gcc 4.9):

Step 2/5 : RUN apt-get update && apt-get -y --no-install-recommends install
cmake g++ git kmod libc6-dev libelf-dev make pkg-config && apt-get clean
 ---> Running in 81e16f5a1f7f
Get:1 http://security.debian.org jessie/updates InRelease [44.9 kB]
Ign http://deb.debian.org jessie InRelease
Get:2 http://deb.debian.org jessie-updates InRelease [16.3 kB]
Get:3 http://deb.debian.org jessie Release.gpg [1652 B]
Get:4 http://deb.debian.org jessie Release [77.3 kB]
Get:5 http://security.debian.org jessie/updates/main amd64 Packages [992 kB]
Ign http://deb.debian.org jessie-updates InRelease
Get:6 http://deb.debian.org jessie-updates/main amd64 Packages [20 B]
Ign http://deb.debian.org jessie Release
Get:7 http://deb.debian.org jessie/main amd64 Packages [9098 kB]
Fetched 10.2 MB in 0s (10.9 MB/s)
Reading package lists...
W: GPG error: http://deb.debian.org jessie-updates InRelease: The following signatures were invalid: KEYEXPIRED 1668891673
W: GPG error: http://deb.debian.org jessie Release: The following signatures were invalid: KEYEXPIRED 1668891673
Reading package lists...
Building dependency tree...
Reading state information...
The following extra packages will be installed:
  binutils cmake-data cpp cpp-4.9 g++-4.9 gcc gcc-4.9 git-man libarchive13
  libasan1 libatomic1 libc-dev-bin libcilkrts5 libcloog-isl4 libcurl3
  libcurl3-gnutls libelf1 liberror-perl libexpat1 libgcc-4.9-dev libgdbm3
  libglib2.0-0 libgomp1 libgssapi-krb5-2 libidn11 libisl10 libitm1
  libk5crypto3 libkeyutils1 libkrb5-3 libkrb5support0 libldap-2.4-2 liblsan0
  liblzo2-2 libmpc3 libmpfr4 libquadmath0 librtmp1 libsasl2-2
  libsasl2-modules-db libssh2-1 libssl1.0.0 libstdc++-4.9-dev libtsan0
  libubsan0 libxml2 linux-libc-dev perl perl-modules
Suggested packages:
  binutils-doc codeblocks eclipse ninja-build cpp-doc gcc-4.9-locales
  g++-multilib g++-4.9-multilib gcc-4.9-doc libstdc++6-4.9-dbg gcc-multilib
  manpages-dev autoconf automake libtool flex bison gdb gcc-doc
  gcc-4.9-multilib libgcc1-dbg libgomp1-dbg libitm1-dbg libatomic1-dbg
  libasan1-dbg liblsan0-dbg libtsan0-dbg libubsan0-dbg libcilkrts5-dbg
  libquadmath0-dbg gettext-base git-daemon-run git-daemon-sysvinit git-doc
  git-el git-email git-gui gitk gitweb git-arch git-cvs git-mediawiki git-svn
  lrzip glibc-doc krb5-doc krb5-user libstdc++-4.9-doc make-doc perl-doc
  libterm-readline-gnu-perl libterm-readline-perl-perl libb-lint-perl
  libcpanplus-dist-build-perl libcpanplus-perl libfile-checktree-perl
  liblog-message-simple-perl liblog-message-perl libobject-accessor-perl
Recommended packages:
  patch less rsync ssh-client ca-certificates libglib2.0-data shared-mime-info
  xdg-user-dirs krb5-locales libsasl2-modules xml-core netbase rename
  libarchive-extract-perl libmodule-pluggable-perl libpod-latex-perl
  libterm-ui-perl libtext-soundex-perl libcgi-pm-perl libmodule-build-perl
  libpackage-constants-perl
The following NEW packages will be installed:
  binutils cmake cmake-data cpp cpp-4.9 g++ g++-4.9 gcc gcc-4.9 git git-man
  kmod libarchive13 libasan1 libatomic1 libc-dev-bin libc6-dev libcilkrts5
  libcloog-isl4 libcurl3 libcurl3-gnutls libelf-dev libelf1 liberror-perl
  libexpat1 libgcc-4.9-dev libgdbm3 libglib2.0-0 libgomp1 libgssapi-krb5-2
  libidn11 libisl10 libitm1 libk5crypto3 libkeyutils1 libkrb5-3
  libkrb5support0 libldap-2.4-2 liblsan0 liblzo2-2 libmpc3 libmpfr4
  libquadmath0 librtmp1 libsasl2-2 libsasl2-modules-db libssh2-1 libssl1.0.0
  libstdc++-4.9-dev libtsan0 libubsan0 libxml2 linux-libc-dev make perl
  perl-modules pkg-config
0 upgraded, 57 newly installed, 0 to remove and 0 not upgraded.
Need to get 60.4 MB of archives.
After this operation, 223 MB of additional disk space will be used.
WARNING: The following packages cannot be authenticated!
  libgdbm3 libkeyutils1 cmake-data liblzo2-2 cmake libisl10 libcloog-isl4
  libmpfr4 libmpc3 kmod binutils cpp gcc g++ liberror-perl make pkg-config
E: There are problems and -y was used without --force-yes

nathan-b commented:
I think this solution is (for me) preferable to ignoring expired keys.

iurly commented:
> I think this solution is (for me) preferable to ignoring expired keys.

Can I get an approval then?
